tangelo/linq/target/backend.py

```diff
@@ -184,6 +184,14 @@ def simulate(self, source_circuit, return_statevector=False, initial_statevector
                 if available.
             initial_statevector(list/array) : A valid statevector in the format
                 supported by the target backend.
+            save_mid_circuit_meas (bool): Save mid-circuit measurement results to
+                self.mid_circuit_meas_freqs. All measurements will be saved to
+                self.all_frequencies.
+                self.all_frequencies will have keys of length m + n_qubits
+                The first m values in each key will be the result of the m MEASURE gates ordered
```

Comment:
I think this format is confusing. Would it be better to have something embedded or separated? Food for thought.
```python
self.all_frequencies = {"measure_0": histogram or dict, "measure_1": histogram or dict}
# or
self.all_frequencies = [{"0": histogram or dict, "1": histogram or dict}, ...] # For each measure gate.
```

I don't really know how to save the frequencies also.

Reply:
OK, I think I see the problem. The second PR will deal with post-selecting on the mid-circuit measurements, which will generate a single histogram based on desired measurement outcomes. This code was meant to only implement the option to save the mid-circuit measurements, as we felt the whole thing was too big for a single PR. The crux of this functionality will be carried by the `marginal_frequencies` method / function.